src/Events/Globals/GlobalsInitializedEvent.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * This file is part of OpenEMR.
+ *
+ * @link https://github.com/openemr/openemr/tree/master
+ * @license https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Events\Globals;
+
+use OpenEMR\Services\UserService;
+use Symfony\Component\EventDispatcher\Event;
+
+/**
+ * Event object for creating custom global settings
+ *
+ * @package OpenEMR\Events
+ * @subpackage Globals
+ * @author Ken Chapple <ken@mi-squared.com>
+ * @copyright Copyright (c) 2019 Ken Chapple <ken@mi-squared.com>
+ */
+class GlobalsInitializedEvent extends Event
+{
+    /**
+     * The global.init event occurs after globals have been set up in globals.inc.php
+     */
+    const EVENT_HANDLE = 'globals.initialized';
+
+    /**
+     * @var null|UserService
+     *
+     */
+    private $userService = null;
+
+    /**
+     * @var array
+     *
+     * The entire globals structure
+     */
+    private $globalsMetadata = [];
+
+    /**
+     * @var array
+     *
+     * User-specific globals
+     */
+    private $userSpecificGlobals = [];
+
+    /**
+     * @var array
+     *
+     *  User specific tabs
+     */
+    private $userSpecificTabs = [];
+
+    /**
+     * GlobalsInitEvent constructor.
+     * @param $userService
+     * @param $GLOBALS_METADATA
+     * @param $USER_SPECIFIC_GLOBALS
+     * @param $USER_SPECIFIC_TABS
+     */
+    public function __construct($userService, $GLOBALS_METADATA, $USER_SPECIFIC_GLOBALS, $USER_SPECIFIC_TABS)
+    {
+        $this->userService = $userService;
+        $this->globalsMetadata = $GLOBALS_METADATA;
+        $this->userSpecificGlobals = $USER_SPECIFIC_GLOBALS;
+        $this->userSpecificTabs = $USER_SPECIFIC_TABS;
+    }
+
+    public function getUserService()
+    {
+        return $this->userService;
+    }
+
+    public function save()
```

Comment:
I think you're mixing behavior. Your confusion is good, though cause maybe I can make this more clear. We're not actually saving global values, we're saving the custom settings into the globals data structure along with default values. Using this code, a developer wouldn't actually be able to change system data, but only the structure. Data isn't actually stored in the DB until user clicks saved. So basically if a module wants to add fields to the global data structure, and have them appear in the Admin > Globals screen, they would add their settings, and "save" them. Hope that makes sense.

I thought about the global service, and I can do that. I didn't because I thought that this would be the only case (in the event) that you'd be doing this stuff, but I might be wrong.